coral/src/app/components/ComplexNativeSelect.tsx

```diff
@@ -0,0 +1,88 @@
+import { NativeSelect, NativeSelectProps } from "@aivenio/aquarium";
+import { omit } from "lodash";
+import { ChangeEvent, useState } from "react";
+
+type ComplexNativeSelectProps<ComplexNativeSelectOptionType> =
```

Comment:
More of a personal preference than functional detail, but when I was reading this code I expected `ComplexNativeSelectOptionType`  to be a proper type instead of a generic. I think the most commonly used convention is to use a single letter types for generics. I would suggest we make it shorter e.g. `T` or `TSelectOption`